backend/main.py

```diff
@@ -478,34 +478,34 @@ async def news_summary(
 
 
 @app.post("/api/v1/news/{id}/upvote")
-def upvote_article(
+def handle_news_article_upvote(
         id,
         db=Depends(session_opener),
         u=Depends(authenticate_user_token),
 ):
-    message = toggle_upvote(id, u.id, db)
+    message = toggle_news_article_upvote(id, u.id, db)
     return {"message": message}
 
 
-def toggle_upvote(n_id, u_id, db):
+def toggle_news_article_upvote(username_id, u_id, db):
     existing_upvote = db.execute(
         select(user_news_association_table).where(
-            user_news_association_table.c.news_articles_id == n_id,
+            user_news_association_table.c.news_articles_id == username_id,
             user_news_association_table.c.user_id == u_id,
         )
     ).scalar()
 
     if existing_upvote:
         delete_stmt = delete(user_news_association_table).where(
-            user_news_association_table.c.news_articles_id == n_id,
+            user_news_association_table.c.news_articles_id == username_id,
```

Comment:
This is not the ID of the username, and this could mislead the meaning of this variable